File: app/src/main/java/it/niedermann/nextcloud/deck/persistence/sync/adapters/db/dao/JoinBoardWithLabelDao.java
package it.niedermann.nextcloud.deck.persistence.sync.adapters.db.dao;

import androidx.room.Dao;
import androidx.room.Query;

import it.niedermann.nextcloud.deck.model.JoinBoardWithLabel;

@Dao
public interface JoinBoardWithLabelDao extends GenericDao<JoinBoardWithLabel> {
    @Query("DELETE FROM joinboardwithlabel WHERE boardId = :localId")
    void deleteByBoardId(long localId);
}

File: data/local/src/main/java/it/niedermann/nextcloud/deck/data/local/dao/JoinBoardWithLabelDao.kt
package it.niedermann.nextcloud.deck.data.local.dao

import androidx.room3.Dao
import androidx.room3.Query
import it.niedermann.nextcloud.deck.data.local.entity.JoinBoardWithLabelEntity
import java.util.concurrent.CompletableFuture

@Dao
interface JoinBoardWithLabelDao : GenericDao<JoinBoardWithLabelEntity> {
    @Query("DELETE FROM JoinBoardWithLabel WHERE boardId = :boardId")
    fun deleteByBoardId(boardId: Long): CompletableFuture<Void?>
}
